tutorials/Jak korzystac z GitHub.md

```diff
@@ -0,0 +1,193 @@
+
+    ![](images/terminal9.jpg)
+* Wpisz komendę `git commit -m "<króki_tekst_opisujący_zapisywane_zmiany>"` i kliknij Enter,
+
+    ![](images/terminal10.jpg)
+* Wpisz komendę `git push` i kliknij Enter,
+
+    ![](images/terminal11.jpg)
+* W celu sprawdzenia, czy coś jeszcze nie zostało zapisane na repozytorium zdalnym, możemy wykonać komendę `git status`, która pokaże, które pliki czekają jeszcze na commit (u mnie widać dodane nowe 3 pliki).
+
+    ![](images/terminal12.jpg)
+
+3. Zapisanie zmian przez Visual Studio Code:
+
+* Sklonuj repozytorium lokalnie tak, jak zostało to pokazane w punkcie 2.1.,
+* Jeśli masz już sklonowane repozytorium, przed wprowadzeniem zmian wykonaj `git pull` tak, jak zostało to pokazane w punkcie 2.2.,
+* Przejdź do swojego brancha za pomocą komendy `git checkout` tak, jak zostało to pokazane w punkcie 2.3.,
+* Otwórz swój plik w Visual Studio Code,
+* Zapisz plik w odpowiednim folderze na repozytorium lokalnym,
+
+    ![](images/visual1.jpg)
+* Wejdź w zakładkę "Source Control"
+
+    ![](images/visual2.jpg)
+* Dodaj pliki, w których chciałbyś zapisać zmiany,
+
+    ![](images/visual3.jpg)
+* Dodaj komentarz opisujący zmiany, które chcesz zapisać,
+
+    ![](images/visual4.jpg)
+* Kliknij "Commit & Push", aby zapisać zmiany na repozytorium zdalnym.
+
+    ![](images/visual5.jpg)
+
+Jeśli wszystkie zmiany dotyczące danego zadania zostały już zapisane na repozytorium, wówczas możemy stworzyć ***pull request***, czyli prośbę o ***merge*** wszystkich zmian z naszego brancha do main. Zanim taki request zostanie zatwierdzony, dodany/zmieniony kod musi przejść ***review***, podczas którego reviewer może mieć pewne sugestie do kodu, komunikując je przez komentarze.
+
+Jak założyć pull request, kiedy skończyliśmy pracę nad danym zadaniem? Tu mamy dwie opcje:
+
+1. Zakładanie pull requesta przez pojawiający się przycisk "Compare & pull request":
+
+* Kliknij przycisk "Compare & pull request" (pojawia się on po zidentyfikowaniu zmian nie wprowadzonych na main),
+
+    ![](images/pull_request5.jpg)
+* Dodaj komentarz jeśli chcesz i kliknij "Create pull request",
+
+    ![](images/pull_request6.jpg)
+* Przypisz osobę, która ma zrobić review w polu "Assignees".
+
+    ![](images/pull_request7.jpg)
+
+2. Zakładanie pull requesta przez zakładkę "Pull requests":
+
+* Wejdź w zakładkę "Pull requests",
+
+    ![](images/pull_request1.jpg)
+* Kliknij "New pull request",
+
+    ![](images/pull_request2.jpg)
+* Wybierz, z jakiego brancha do jakiego ma być merge (zazwyczaj będziemy merge'ować do main),
+
+    ![](images/pull_request3.jpg)
+* Kliknij "Create pull request",
+
+    ![](images/pull_request4.jpg)
+* Dodaj komentarz jeśli chcesz i kliknij "Create pull request",
+
+    ![](images/pull_request6.jpg)
+* Przypisz osobę, która ma zrobić review w polu "Assignees".
+
+    ![](images/pull_request7.jpg)
+
```

Comment:
Bym poprosił o zapisanie sekcji o złotych zasadach.
* Ustalamy kto robi merge oraz rozwiązuje konflikty
* Dobra praktyka to rozwiązać konflikty zanim przekaże się kod do review
* Komentarze kończy (resolve) osoba która je utworzyła.
* Nigdy nie mergujemy bez uzyskania approve
* etc

Reply:
dodane

Reply:
'Nigdy nie używamy opcji "Close pull request" ' zamieniłbym na "Nie używamy ... pull request. Oznacza ona rezygnację z chęci zakończenia zadania lub, że problem został rozwiązany gdzie indziej".

Reply:
"Merge branchy zawsze robi team lead." Nie zawsze. Zamieniłbym na 
"Zespoły przyjmują stałą zasadę co do tego kto odpowiada za merge danego pull requesta. Np. obecnie wykonuje to dr Kowalski, ale docelowo chcemy aby nasz lider to robił."

Reply:
zmienione